plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseDistributedQueries.java

```diff
@@ -0,0 +1,220 @@
+
+    @Override
+    protected boolean supportsCommentOnTable()
+    {
+        return false;
+    }
+
+    @Override
+    public void testCommentColumn()
+    {
+        // currently does not support
+        // TODO
+        assertThatThrownBy(super::testCommentColumn)
+                .hasMessage("This connector does not support setting column comments");
+        throw new SkipException("Unsupported");
+    }
+
+    @Override
+    public void testRenameTable()
+    {
+        String tableName = "test_rename_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (x int)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (123)", 1);
+
+        String renamedTable = "test_rename_new_" + randomTableSuffix();
+        assertUpdate("ALTER TABLE " + tableName + " RENAME TO " + renamedTable);
+        assertQuery("SELECT x FROM " + renamedTable, "VALUES 123");
+
+        String testExistsTableName = "test_rename_new_exists_" + randomTableSuffix();
+        assertUpdate("ALTER TABLE IF EXISTS " + renamedTable + " RENAME TO " + testExistsTableName);
+        assertQuery("SELECT x FROM " + testExistsTableName, "VALUES 123");
+
+        String uppercaseName = "TEST_RENAME_" + randomTableSuffix(); // Test an upper-case, not delimited identifier
+        assertUpdate("ALTER TABLE " + testExistsTableName + " RENAME TO " + uppercaseName);
+        assertQuery("SELECT x FROM " + uppercaseName.toLowerCase(ENGLISH), // Ensure select allows for lower-case, not
+                // delimited identifier
+                "VALUES 123");
+
+        assertUpdate("DROP TABLE " + uppercaseName);
+
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
+        assertFalse(getQueryRunner().tableExists(getSession(), renamedTable));
+
+        assertUpdate("ALTER TABLE IF EXISTS " + tableName + " RENAME TO " + renamedTable);
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
+        assertFalse(getQueryRunner().tableExists(getSession(), renamedTable));
+    }
+
+    @Override
+    public void testRenameColumn()
+    {
+        // TODO
+    }
+
+    @Override
+    public void testDropColumn()
+    {
+        // TODO
+    }
+
+    @Override
+    public void testAddColumn()
+    {
+        // TODO
+        throw new SkipException("Unsupported");
+    }
+
+    @Override
+    public void testInsertArray()
+    {
+        // TODO
+        assertThatThrownBy(super::testInsertArray)
+                .hasMessage("not supported");
+        throw new SkipException("Unsupported");
+    }
+
+    @Override
+    public void testInsertForDefaultColumn()
+    {
+        // TODO
+        assertThatThrownBy(super::testInsertForDefaultColumn)
+                .hasMessage("NULL value not allowed for NOT NULL column: col_nullable");
+        throw new SkipException("Unsupported");
+    }
+
+    @Override
+    @Test(dataProvider = "testColumnNameDataProvider")
+    public void testColumnName(String columnName)
```

Comment:
Does clickhouse has case sensitive column names? If so, this test should be easy to enable. Take a look at the PostgreSQL connector's test for an example.

Reply:
Yes, clickhouse has case sensitive column names